src/services/oracle.ts

```diff
@@ -37,36 +36,18 @@ export const OracleAbi = [
  */
 export class OracleService<T extends ChainId> extends ContractService<T> {
   static abi = OracleAbi;
+  static contractId = ContractAddressId.oracle;
 
-  constructor(chainId: T, ctx: Context) {
-    super(ctx.addresses.oracle ?? OracleService.addressByChain(chainId), chainId, ctx);
+  get contract() {
+    return super._getContract(OracleService.abi, OracleService.contractId, this.ctx);
```

Comment:
btw I think we can use `this._getContract` instead of super since it already inherits from ContractService, but I might be making a mistake here

Reply:
It's the same unless `this` overwrites the `super` implementation but I fixed it for clarity